bladeRF-cli: Can now specify multiple commands separated by semicolon

jenselo opened the pull request:
bladeRF-cli now supports multiple commands separated by semicolon, as requested in this issue https://github.com/Nuand/bladeRF/issues/345. If done in non-interactive mode, the command list has to be surrounded by quotes, ie. bladeRF-cli -e "probe;info" since bash also separates commands on semicolon.

jynik commented:
@jenselo,  I know we discussed this briefly on IRC a while back and am just trying to remember where we left off.  Below is what I believe the summary was, please correct me if I'm wrong.

- I had changed my mind and wanted this change to address semicolons within quotes before merging it.

- We discussed moving the breaking up of the line to `input_loop()` in `input.c` as this is where input is being handled (from either a script, cmdline, or interactive intput) and being dispatched to the command handler.

- We had discussed perhaps using a simpler approach that just scans the input line, inserts a `'\0'` and keeps a ptr to the next string, if any remain (thereby avoiding some of the dynamic memory alloc operations, since we're free to touch the input line).

jenselo commented:
@jynik Yep, that's what I remember as well. I have moved the code to input_loop, and added a separate function for splitting it to input.c. I have not looked into replacing ';' with '\0', since I've been busy with final exams, but I can take a look at it maybe tomorrow or definitely next week.

jynik commented:
@jenselo Thanks - was just reviewing pull requests and keep my head on straight, no rush at all.

Best of luck with your remaining exams! :)

jynik commented:
Just removing the 2015.02 milestone. We can add new functionality again in the 2015.03-rc1.

jenselo commented:
Now '\0' is inserted into the input line  the way discussed. Although there may be som unneccesary commit (like merges) in the commit log.

jynik commented:
It's fine with me if you want to rebase against master and force a push to your branch. It saves me the trouble of having to rebase...

I think the patch just needs one more minor iteration before I can pull it into a staging branch for the next RC1.  I believe `char *commands[max_cmds];` will fail to build with the MSVC compiler, as it tends not to support C99 features, such as variable-length arrays. 

Even if it compiles fine, we don't need an array here -- we just need to store a pointer to the start of the command string, since we only care about the current command string. (It's not as if we're going back to previous commands and need to keep those pointers around.)

Thanks!

jynik commented:
Merged in b6ba3b20d669b1f8373974fe4ee3df253c92a020.

Did a little massaging in bc1f6f2106dc727a9433948f8ec307571fe60ddb for MSVC build and readability.
